csrc/string_vector.h

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#ifndef STRING_VECTOR_H
+#define STRING_VECTOR_H
+
+#include "compiler.h"
+
+#include <cstdio>
+#include <cstdlib>
+#include <pthread.h>
+#include <string>
+#include <vector>
+
+namespace AmazonCorrettoCryptoProvider {
+
+class ConcurrentStringVector {
+private:
+    std::vector<std::string> vec;
+    mutable pthread_rwlock_t lock;
```

Comment:
I originally wrote this using `std::mutex`, but GCC 4.1.2 apparently doesn't support that, `std::lock_guard`, or std::unique_lock`. So, I [rewrote it](https://github.com/corretto/amazon-corretto-crypto-provider/pull/433/commits/13c9f3cac18352d9dfa9b898bdb2a42a9c1e82f5) with pthread. Relevant man page [here](https://linux.die.net/man/3/pthread_rwlock_init).